playback/src/player.rs

```diff
@@ -723,7 +723,19 @@ impl PlayerTrackLoader {
 
         let audio_file = Subfile::new(decrypted_file, 0xa7);
 
-        let mut decoder = match VorbisDecoder::new(audio_file) {
+        let result = if self.config.passthrough {
+            match PassthroughDecoder::new(audio_file) {
+                Ok(result) => Ok(Box::new(result) as Decoder),
+                Err(e) => Err(AudioError::PassthroughError(e)),
+            }
+        } else {
+            match VorbisDecoder::new(audio_file) {
+                Ok(result) => Ok(Box::new(result) as Decoder),
+                Err(e) => Err(AudioError::VorbisError(e)),
+            }
+        };
+
```

Comment:
I'm not sure this is the best option but my Rust "rookieness" really hurts here

Reply:
I think it's fine. You _could_ use `Result::map` and  `map_err` instead of `match`, but I don't think it's a big win.

Reply:
thx!